godot-core/src/engine/translate.rs

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Macros for translation.
+
+pub use crate::{tr, tr_n};
+
+/// A convenience macro for using the [`Object::tr()`](crate::engine::Object::tr()) and [`Object::tr_ex()`](crate::engine::Object::tr_ex())
+///  methods.
+///
+/// The methods are called from the [`Engine`](crate::engine::Engine) singleton.
+///
+/// Takes a format string literal, with optional arguments. Optionally, `context` for potentially ambiguous words can be
+/// added before the format arguments, separated with a `;`.
+///
+/// For maximum flexibility and minimum ambiguity, prefer inlined, positional, or named arguments in the format strings:
+/// ```no_run
+/// # use godot_core::tr;
+/// # struct Foo { x: i32, y: i32 };
+/// # impl std::fmt::Display for Foo {
+/// #    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+/// #        write!(f, "")
+/// #    }
+/// # }
+/// # let a = Foo { x: 0, y: 0 };
+/// # let b = Foo { x: 0, y: 0 };
+/// # let context = "context";
+/// // Good
+/// tr!(context; "{a} is a {b}"); // inlined, with context
+/// tr!("{0} is a {1}", a, b); // positional, without context
+/// tr!("{c} is a {d}", c = a.x, d = b.y); // named (inlining not possible here)
+///
+/// // Not so good, much more fragile
+/// // and does not easily allow ordering arguments differently in other translations.
+/// tr!("{} is a {}", a, b);
+/// ```
+/// see also: <https://docs.godotengine.org/en/stable/tutorials/i18n/internationalizing_games.html#translation-contexts>
+#[macro_export]
+macro_rules! tr {
+    ($fmt:literal $(, $($args:tt)*)?) => {
+        $crate::engine::Engine::singleton()
+            .tr(format!($fmt $(, $($args)*)?).into())
+    };
+
+    ($cxt:expr; $fmt:literal $(, $($args:tt)*)?) => {
```

Comment:
The abbreviation would be `ctx`, but I'd write `context` instead. 
It appears [in the docs](https://godot-rust.github.io/docs/gdext/pr-596/godot/engine/translate/macro.tr_n.html), so that's a bit clearer.

(same for the other macro)

Reply:
D'oh. Good catch on the incorrect abbreviation, but yeah I'll just change it to `context`.